ingestion/src/metadata/ingestion/source/dashboard/tableau/metadata.py

```diff
@@ -290,6 +251,81 @@ def yield_tag(self, *_, **__) -> OMetaTagAndClassification:
                     logger.debug(traceback.format_exc())
                     logger.error(f"Error ingesting tag [{tag}]: {err}")
 
+    def get_column_info(self, data_model: TableauDataModel) -> Optional[List[Any]]:
+        """
+        get columns details for Data Model
+        """
+        datasource_columns = []
+        for column in data_model.datasourceFields:
+            parsed_string = {
+                "dataTypeDisplay": column.remoteField.dataType,
```

Comment:
Documentation says:
```
# Reference to a field from a published data source. This property only exists on
# Fields that are in an embedded data source with a connection to a published data
# source.
```
does it mean that `column.remoteField` can be `None`?

Reply:
datasourceFields list will be an empty list in that case, so we will never iterate over it 👍